### PR TITLE
feat(accident-notification): connect new api for accident-notification

### PR DESCRIPTION
**Pull request by jonnigs**
## What

Connect a new client for accident notifications and stop using the old xml client

## Why

To modernize

## Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes

## Summary by CodeRabbit

- **New Features**
	- Enhanced accident notification status retrieval by incorporating user context for personalized data handling.
	- Streamlined the `ApplicationStatus` component logic, improving readability and maintainability through the introduction of utility functions.
	- Added new functions for improved attachment management, enabling better handling of accident report submissions and associated data.

- **Bug Fixes**
	- Updated error handling in the `ApplicationStatus` component to provide clearer feedback when data is unavailable.

- **Documentation**
	- Improved clarity in method signatures and component props for better developer understanding.

- **Chores**
	- Removed unused imports and refactored component logic to reduce redundancy.

### CODE REVIEW

File: libs/application/template-api-modules/src/lib/modules/templates/accident-notification/accident-notification-v2.utils.ts [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+
+const whoIsTheNotificationForToDTO = (who: WhoIsTheNotificationForEnum) => {
+  return (
+    reportingForMap[who] ||
+    MinarsidurAPIModelsAccidentReportsReporterDTOReportingForEnum.NUMBER_1
+  )
+}
+
+/*
+ * type can be 4: Íþróttaslys, 6: Vinnuslys, 7: Heimilistrygging, 8: Björgunarmenn, 9: Nemendur við iðnnám
```
Comment (norda-gunni) [inline libs/application/template-api-modules/src/lib/modules/templates/accident-notification/accident-notification-v2.utils.ts:65]: Consider making this line a bit more readable, something like
```suggestion
 * type can be one of the following:
    4: Íþróttaslys, 
    6: Vinnuslys, 
    7: Heimilistrygging, 
    8: Björgunarmenn, 
    9: Nemendur við iðnnám
```

File: libs/application/template-api-modules/src/lib/modules/templates/accident-notification/accident-notification-v2.utils.ts [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+
+//     if (StudiesAccidentTypeEnum.VOCATIONALEDUCATION === studiesAccidentType) {
+//       subtype = 7
+//     }
+
+//     return { type: 9, subtype }
+//   }
+
+//   return baseType
+// }
```
Comment (norda-gunni) [inline libs/application/template-api-modules/src/lib/modules/templates/accident-notification/accident-notification-v2.utils.ts:189]: Clean up commented lines?

**Review by norda-gunni — COMMENTED** [review]
Just some minor stuff

File: libs/application/template-api-modules/src/lib/modules/templates/accident-notification/accident-notification-v2.utils.ts [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+    description: accidentDetails.descriptionOfAccident ?? '',
+    fatal: fatal === 'yes',
+    location: accidentLocation,
+    locationDescription: accidentDetails.descriptionOfAccident ?? '',
+    symptoms: accidentDetails.accidentSymptoms ?? '',
+    dateTimeOfDoctorVisit: accidentDetails.dateOfDoctorVisit
+      ? new Date(accidentDetails.dateOfDoctorVisit)
+      : new Date(),
+    // dockName: null, // Not in the application, but should it?
+    // dockGps: null, // Not in the application, but should it?
```
[the remark this answers is not in the document]
Comment (jonnigs) [reply libs/application/template-api-modules/src/lib/modules/templates/accident-notification/accident-notification-v2.utils.ts:257]: yes

**Review by norda-gunni — APPROVED** [review]
LGTM

**Review by thordurhhh — APPROVED** [review]
Client changes LGTM 🚀

**Pull request merged**